app/views/posts/index.html.erb

```diff
@@ -1,7 +1,37 @@
-<h1>Here is a list of posts for user <%= @user.id %></h1>
-<% if @posts.exists? %>
-  <% @posts.each do |post| %>
-    <h3>Title: <%= post.title %></h3>
-    <p>Text: <%= post.text %></p>
-  <% end %>
-<% end %>
\ No newline at end of file
+<main id="posts" class="page">
+  <%# Home button start %>
+  <div id="user-page">
+    <div class="btn-container">
+      <%= link_to "Home", root_path %>
+    </div>
+    <%# End of Home button %>
+    <%# User profile start %>
+    <div class="user-container posts">
+      <div class="user-profile-pic">
+        <img src="<%= @user.photo %>" alt="profile-picture">
+      </div>
+      <div class="user-info">
+        <h3><%= @user.name %></h3>
+        <p class="posts-counter">Number of posts: <%= @user.posts_counter || 0 %></p>
+      </div>
+    </div>
+    <%# End of user profile %>
+    <article>
+    <% @posts.each do |post| %>
+      <%= link_to user_post_path(@user, post) do %>
+        <div class="post-container">
+          <h3 class="post-title"><%= post.title %></h3>
+          <div class="post-details">
+            <p class="details"><%= post.text[0..100]... %></p>
+          </div>
+          <div class="comment-like-container">
+            <p class="comment-likes-data">
+              Comment: <%= post.comments.count %>, Likes: <%= post.likes.count %>
+            </p>
+          </div>
+        </div>
+      <% end %>
+    <% end %>
+  </article>
```

Comment:
Kindly use `post.recent_comments` (in your case `most_recent_comments`) to render 5 most recent comments in the `app/views/posts/index.html.erb` page.

For example:
```ruby
 <% post.recent_comments.each do |comment| %>
  <p><%= comment.author.name %>: <%= comment.text %></p> 
 <% end %>
```

- Users should see `5 recent comments` for each post.

Reply:
@KDlamini Thanks for the review and I just forgot to include that. thanks for mentioning it.